sys/include/net/gnrc/coap.h

```diff
@@ -7,382 +7,17 @@
- *
- * @param[in] pdu Request metadata
- * @param[in] payload_len Length of the payload, or 0 if none
- * @param[in] format Format code for the payload; use COAP_FORMAT_NONE if not
- *                   specified
- *
- * @return size of the PDU
- * @return < 0 on error
- */
-ssize_t gcoap_finish(coap_pkt_t *pdu, size_t payload_len, unsigned format);
-
-/**
- *  @brief Writes a complete CoAP request PDU when there is not a payload.
- *
- * @param[in] pdu Request metadata
- * @param[in] buf Buffer containing the PDU
- * @param[in] len Length of the buffer
- * @param[in] code Request code
- * @param[in] path Resource path
- *
- * @return size of the PDU within the buffer
- * @return < 0 on error
- */
-static inline ssize_t gcoap_request(coap_pkt_t *pdu, uint8_t *buf, size_t len,
-                                                                   unsigned code,
-                                                                   char *path)
-{
-    return (gcoap_req_init(pdu, buf, len, code, path) == 0)
-                ? gcoap_finish(pdu, 0, COAP_FORMAT_NONE)
-                : -1;
-}
-
-/**
- * @brief  Sends a buffer containing a CoAP request to the provided host/port.
- *
- * @param[in] buf Buffer containing the PDU
- * @param[in] len Length of the buffer
- * @param[in] addr Destination for the packet
- * @param[in] port Port at the destination
- * @param[in] resp_handler Callback when response received
- *
- * @return length of the packet
- * @return 0 if cannot send
- */
-size_t gcoap_req_send(uint8_t *buf, size_t len, ipv6_addr_t *addr, uint16_t port,
-                                                gcoap_resp_handler_t resp_handler);
-
-/**
- * @brief  Initializes a CoAP response packet on a buffer.
- *
- * Initializes payload location within the buffer based on packet setup.
- *
- * @param[in] pdu Response metadata
- * @param[in] buf Buffer containing the PDU
- * @param[in] len Length of the buffer
- * @param[in] code Response code
- *
- * @return 0 on success
- * @return < 0 on error
- */
-int gcoap_resp_init(coap_pkt_t *pdu, uint8_t *buf, size_t len, unsigned code);
-
-/**
- * @brief  Writes a complete CoAP response PDU when there is no payload.
- *
- * @param[in] pdu Response metadata
- * @param[in] buf Buffer containing the PDU
- * @param[in] len Length of the buffer
- * @param[in] code Response code
- *
- * @return size of the PDU within the buffer
- * @return < 0 on error
- */
-static inline ssize_t gcoap_response(coap_pkt_t *pdu, uint8_t *buf, size_t len,
-                                                                    unsigned code)
-{
-    return (gcoap_resp_init(pdu, buf, len, code) == 0)
-                ? gcoap_finish(pdu, 0, COAP_FORMAT_NONE)
-                : -1;
-}
-
-/**
- * @brief Provides important operational statistics.
- *
- * Useful for monitoring.
- *
- * @param[out] open_reqs Count of unanswered requests
- */
-void gcoap_op_state(uint8_t *open_reqs);
-
-#ifdef __cplusplus
-}
-#endif
+#include "net/gcoap.h"
```

Comment:
No, coap.h is not required any longer -- it's only there for backwards compatibility. I didn't want to break code for existing users out of the box. I included a 'deprecated' comment above, but I'm happy to remove the file. I don't expect that there are hordes of gcoap users, but is there a policy on when to deprecate?